### PR TITLE
Support writing to uninitialized memory

### PR DESCRIPTION
**Pull request by james7132**
This PR implements BufferMut for slices and Vecs of `MaybeUninit<u8>`. This also implements BytesVecExt provide an implementation that extends the Vec without zeroing out the memory, explicitly for `MaybeUninit<u8>`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,6 +133,30 @@ assert_eq!(offsets, [0, 64, 192]);
...
+
+let mut buffer = DynamicStorageBuffer::new_with_alignment(&mut byte_buffer, 64);
+let offsets = [
+    buffer.write(&[5.; 10]).unwrap(),
+    buffer.write(&vec![3u32; 20]).unwrap(),
+    buffer.write(&glam::Vec3::ONE).unwrap(),
+];
+
+// SAFETY: Vec<u8> and Vec<MaybeUninit<u8>> share the same layout.
+let byte_buffer: Vec<u8> = unsafe { std::mem::transmute(byte_buffer) };
```
Comment (SkiFire13) [inline README.md:152]: This relies on the unspecified layout of `Vec` and might result in UB if the compiler ever reorders the fields of one type but not the other. The proper way would be to use `Vec::from_raw_parts`. See the second to last example of `std::mem::transmute`'s documentation.

Comment (james7132) [reply README.md:152]: I'll switch this to a boxed slice then since Vec::into_raw_parts is unstable and this is shopped to be an example.

**Review by teoxoy — COMMENTED** [review]
Looks good besides the concern above!

File: README.md (hunk 1 above)
Comment (james7132) [reply README.md]: Done!

**Review by SkiFire13 — APPROVED** [review]

**Pull request merged**